### PR TITLE
[Merged by Bors] - chore: split Algebra.Group.Nat

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Ruben-VandeVelde** [comment]
|Files|Import difference|
|-|-|
| `Mathlib.Algebra.Group.Nat`|-204|
| `Mathlib.Tactic.Sat.FromLRAT`|-105|
| `Mathlib.Algebra.Ring.Nat`|-102|
| `Mathlib.Data.Nat.PSub`|-79|
| `Mathlib.Data.Nat.Bits`|-78|
| `Mathlib.Data.Nat.Size`|-72|
| `Mathlib.Data.List.SplitLengths`|-44|
| `Mathlib.Data.Ordmap.Ordnode`|-39|
| `Mathlib.Algebra.Order.Group.Nat`|-32|
| `Mathlib.Data.Set.Enumerate`|-31|
| `Mathlib.Data.Nat.Cast.WithTop`|-29|
| `Mathlib.Tactic.NormNum.Core` `Mathlib.Tactic.NormNum.Result`|-26|
| `Mathlib.Data.String.Basic`|-24|
| `Mathlib.Data.Nat.GCD.Basic`|-20|
| 3 files |-18|
| 25 files |-15|
| `Mathlib.Data.Multiset.OrderedMonoid`|-12|
| `Mathlib.Data.Nat.Prime.Infinite`|-7|
| `Mathlib.Tactic.CancelDenoms.Core`|-6|
| `Mathlib.Data.Nat.Bitwise`|-5|
| 25 files |-3|
| 13 files |-2|
| 29 files |1|
| 3982 files |2|
| `Mathlib.Algebra.Group.Nat.Basic`|71 (new file)|
| `Mathlib.Algebra.Group.Nat.Units`|105 (new file)|
| `Mathlib.Algebra.Group.Nat.TypeTags`|149 (new file)|
| `Mathlib.Algebra.Group.Nat.Even`|204 (new file)|

File: Mathlib/Data/Multiset/Basic.lean [hunk 1]
```diff
@@ -3,14 +3,14 @@ Copyright (c) 2015 Microsoft Corporation. All rights reserved.
...
 import Mathlib.Algebra.Order.Sub.Unbundled.Basic
+import Mathlib.Data.List.Perm.Basic
+import Mathlib.Data.List.Perm.Lattice
 import Mathlib.Data.List.Perm.Subperm
 import Mathlib.Data.Set.List
 import Mathlib.Order.Hom.Basic
-import Mathlib.Data.List.Perm.Lattice
-import Mathlib.Data.List.Perm.Basic
-
 /-!
```
Comment (Vierkantor) [inline Mathlib/Data/Multiset/Basic.lean:14]: A newline got deleted here, which we usually keep. (right?)
```suggestion

/-!
```

**Review by Vierkantor — COMMENTED** [review]
Ooh, nice work! I looked at this file and it always felt not quite worth the effort, but you're right that it has some good downstream impacts :D

bors d+

**Comment by Ruben-VandeVelde** [comment]
Thanks!

bors r+

**Pull request closed without merging**